### ISSUE TITLE
[Bug]: Subtitle placement issue

### ISSUE THREAD

**Issue by ArthurKun21** [FreeTubeApp/FreeTube#6217]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

1. Go to a video like this https://youtu.be/haDjmBT9tu4?t=12
2. Turn on the subtitle

    ![image](https://github.com/user-attachments/assets/f305c228-c3cd-4050-b0bd-a6550e22dd0e)

3. Expect the subtitle to be centered or not placed on either left or right side of the screen

### Expected Behavior

Expect that the subtitle was placed on the center of the screen.

### Issue Labels

accessibility issue, inconsistent behavior, visual bug

### FreeTube Version

75b8a358fde128276b8eb8406657361a971b3979

### Operating System Version

Windows 11 23H2

### Installation Method

.exe

### Primary API used

Local API

### Additional Information

Auto-generated caption also sometimes have caption problems but it was mostly overflowing.

---

This ones you can clearly see the problems with the captions

Sample Images

https://youtu.be/haDjmBT9tu4?t=8

![image](https://github.com/user-attachments/assets/8bfad217-4602-490b-8905-15e01431dcf5)

https://youtu.be/haDjmBT9tu4?t=9

![image](https://github.com/user-attachments/assets/f2044d4f-6174-49d3-be87-046aad70b0f8)

https://youtu.be/haDjmBT9tu4?t=12

![image](https://github.com/user-attachments/assets/965ae490-0627-435e-9cc3-0409dc6f79e6)

But it is not like it always fails though, sometimes it is good enough like this

https://youtu.be/haDjmBT9tu4?t=36

![image](https://github.com/user-attachments/assets/90613b6a-e0ef-485e-b1c5-9ee9ceea9c6d)

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by ArthurKun21** [comment]
The video's subtitle looks like got fixed. 

---

Another example I found.

https://youtu.be/5ADaIvhhcCU?t=86

![image](https://github.com/user-attachments/assets/0227ba6b-c110-4bb1-97f7-3fad224accba)

![image](https://github.com/user-attachments/assets/6005b204-5aa4-45df-9975-20f7edbda5b0)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Im able to reproduce this on v0.22.0-nightly-5290 Beta. On YT they use custom subtitles so i think its maybe related to that.

![image](https://github.com/user-attachments/assets/41700889-b7bc-41e7-9c8a-de0e48593e54)

**Comment by rehashedsalt** [comment]
I too get this from time to time, but I'd like to link this specific video from f4mi for whoever might attempt to implement a solution:

[Poisoning AI with ".аss" subtitles](https://youtu.be/NEDFUjqA1s8)

She very thoroughly abuses the YouTube caption system, both by employing fancy styling and, if the UK English captions are selected, by rendering captions offscreen during normal playback and black-on-black captions during scene transitions. The video also documents her methods and tooling used for doing so, conveniently.

I make no comment on whether what she's doing is a good idea, but the video would serve as a great test case.

**Comment by codebloodedkaroum** [comment]
Thanks for reporting this! I was able to reproduce the behavior you mentioned by visiting the video link and enabling captions. From what I observed, the captions are displaying clearly and are fully readable.

It seems that the placement of the subtitles (not being centered) may have been configured intentionally by the video owner or as part of how the captions were uploaded to YouTube. FreeTube appears to be rendering them as they are provided.

That said, I agree that consistent and centered subtitles would improve accessibility, especially if the placement isn't intentional. Perhaps adding an option to override caption positioning in the player settings could be a helpful improvement.

Thanks again for bringing this up!


https://github.com/user-attachments/assets/e6c5c119-6e8b-459a-809d-3f0098bd0bc6

**Comment by PostScriptReal** [comment]
Alright, I've got a lot to add to this issue. To confirm what other people have been saying, yes, this only affects subtitles that use the .ytt format (.ass subtitles converted by YT Sub Converter), subtitles made from anything else (with the exception of .stl subtitles, though there's really only one or two issues) other than the .ytt format isn't affected. I've got a **big** list of all of the issues that I've noticed in the subtitles FreeTube presents vs. how YouTube presents it (The following examples have been taken from my captioning work on my video '[I Finally played REPO!](https://youtu.be/CSn8-mXqBfk)', if you're testing it for yourself, make sure 'English (Australia)' captions are selected.):

### Positioning bugs
#### Centring
Just like how everyone's saying, subtitles don't get centred properly, currently the problem has been mitigated a bit, but there's still issues. Any subtitle that exceeds a certain amount of characters will go off to the right, I don't know the exact amount that's needed for it to happen, but it's somewhere in-between 25-28 characters.
#### Alignment & Positioning
When a subtitle is being displayed in a different spot (bottom-left, bottom-right, centre, etc.) there's usually not a problem, however the same issue with centring happens when the alignment is at the centre or top-centre as well. 

What other people don't realise though, is that subtitles that use alignment & positioning (\an and \pos) to make the subtitle higher or lower on the y-axis (or a different place on the x-axis that can't be achieved with alignment alone), the way Freetube displays it is pretty hit-or-miss. Sometimes it will display correctly, but other times, like in the example below, it doesn't:
(FREETUBE ON LEFT, YOUTUBE ON RIGHT!)

<img width="400" height="225" alt="Positioning error on FreeTube" src="https://github.com/user-attachments/assets/74745fc8-dc69-4f80-8975-8a685d4e1c11" />
<img width="400" height="225" alt="Correct Positioning on YouTube" src="https://github.com/user-attachments/assets/c142546c-e518-400c-95fc-ed22216b2e79" />

#### Moving Subtitles
Whenever there's a subtitle that moves on either axis (utilising the \move tag), it either doesn't move at all, moves on one axis only, or moves on all axes (in a broken way). Here's a few examples of this.

EXAMPLE 1:
![Image](https://github.com/user-attachments/assets/013bffda-6df2-4e9d-bc89-4345215d4564)![Image](https://github.com/user-attachments/assets/59b0adbb-9272-4a4e-98b1-88599e24e38c)

EXAMPLE 2:

![Image](https://github.com/user-attachments/assets/60db6226-f314-4c04-9b50-7d8f20a6ceb6)![Image](https://github.com/user-attachments/assets/a008c3f0-db39-4cfc-9c0d-c51670e2c988)

EXAMPLE 3:

![Image](https://github.com/user-attachments/assets/29d61003-d2df-4175-8344-46b9ba7b3b8c)![Image](https://github.com/user-attachments/assets/ce635b78-31a0-41ac-85d6-b16bb90da31b)

### Colouring issues
If you looked at the last example for the moving subtitles section, you may have noticed that the cyan colouring for the text didn't display on FreeTube. For whatever reason, FreeTube is allergic to displaying certain text colours. This issue doesn't exclusively happen with .ytt subtitles as this also happens with another officially supported format, EBU STL (subtitle format for European Union TV broadcasts). For .stl subtitles, FreeTube is able to display most of the text colours in the limited palette that the format allows you to use, however, the colour green doesn't get displayed. I have an example from a video on my main channel called '[Exploring the 2003 Half-Life 2 Beta Campaign: Part 1 - Canals](https://www.youtube.com/watch?v=dNUi8DOo_jM)', which shows this discrepancy:

<img width="400" height="202" alt="Image" src="https://github.com/user-attachments/assets/3e538958-4287-4a22-9e51-fa51d842b978" />
<img width="400" height="205" alt="Image" src="https://github.com/user-attachments/assets/2effe2fa-dcaa-4d61-9b8e-7bb2f20715d9" />

### Issues with miscellaneous effects
#### Font Sizing
When there's a subtitle that's being displayed at a smaller or bigger font size, FreeTube doesn't reflect that change. In the subtitles I did for my REPO video, I had set the font size to be smaller based on how far away the speaker was relatively to me, so you can see what I'm talking about in the following example:

<img width="400" height="225" alt="Font size on FreeTube" src="https://github.com/user-attachments/assets/f505f355-1ac1-4507-92c8-38be9970a33f" />
<img width="400" height="225" alt="Font size on YouTube" src="https://github.com/user-attachments/assets/4aa51979-5b74-4b9b-8f6b-f28248b75505" />

#### Animated Transform
Animated Transform effects (such as changing scale, colour, etc.) don't work on FreeTube, in the same video I mentioned earlier, I had made it so when a speaker goes closer to me during the spoken line, the text starts small and then goes back to normal size to indicate this. This effect doesn't display on FreeTube, the following example shows this:

![Image](https://github.com/user-attachments/assets/b307155c-680c-44e3-93d6-824c631cb137)![Image](https://github.com/user-attachments/assets/2ab94d36-201f-4e98-857f-a804838cc009)


There could be more issues that I haven't noticed, but that's all I have!

**Comment by absidue** [comment]
@PostScriptReal This issue is only for subtitles overflowing off the side of the player or being scrunched up to the edge and not being readable. FreeTube's subtitle display is not intended to match YouTube's, it is not implemented and as such is not considered a bug (a bug is something that is broken, wanting a feature that doesn't exist is not a bug).

YouTube and Freetube use completely different players so their handling of subtitles is completely different, it is impossible for us to us to use the same player as YouTube because YouTube has a completely custom one, theirs is not an off the shelf one like the one FreeTube uses. While YouTube does offer subtitles in standard formats, they only include a small amount of information in them compared to the custom format that they use in their own players. Even if we added a translation layer from YouTube's custom format to one of the standard formats supported by the subtitle handling in the video player, it still wouldn't look the same as on YouTube because the subtitle displaying is still completely different, so we would have to implement a completely custom subtitle displayer too and everything together is a massive amount of work.

So to summarise what you are saying is not a bug and has nothing to do with this issue, it's a feature request that requires a massive amount of work.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ArthurKun21 are you still experiencing this on normal videos. Videos that dont use custom subtitles like the mkbhd one you linked

**Comment by ArthurKun21** [comment]
I don't have a specific video right now, but yes still experiencing it.

**Comment by PostScriptReal** [comment]
Hmm. @ArthurKun21 Can you link a video just for extra confirmation? I've only ever had this issue on custom subtitles, i.e. .ytt/.ass subtitles. I recently made [a video](https://youtu.be/ZtNEfsN9-LY) that uses normal .srt captions and they work completely fine when I load it up on FreeTube. The other example you posted since the MKBHD video have used custom subtitles, as it has two captions with top and bottom-center alignment. It'd be nice if you could give us another example if you can, thank you.
